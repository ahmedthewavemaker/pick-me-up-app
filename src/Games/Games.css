
*{
    box-sizing: border-box;
}

.gameview{
    display: flex;
    flex-direction: row;
}

.gametime{
 background-color: rgba(0, 0, 0, 0.8);
 margin: auto;
 padding: 15px;
}