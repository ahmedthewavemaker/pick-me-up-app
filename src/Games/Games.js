import React, { useState, useEffect, useCallback } from 'react';
import './Games.css'
import config from '../config'
import moment from 'moment'


function Games() {
    const getGames = () => {
        return fetch(`${config.API_ENDPOINT}/games`)

            .then(res => res.json())
            .then(games => {
                setGames(games)


            })
            .catch(res => {
                console.log(res)
                setError(error)
            })
    }
    const [error, setError] = useState(null)
    const callback = useCallback(getGames, [error])
    
    const [games, setGames] = useState([])
    useEffect(() => {
       callback()
    }, [callback])

    //3 functions of the requests (GET and DELETE)


    const deleteGames = (id) => {
        fetch(`${config.API_ENDPOINT}/games/${id}`, {
            method: 'DELETE',

        })

            .then(games => {
                getGames()
            })


    }

    return (
        <div className='gameview' >
            {games.map(game => (<div className='gametime' key={`${game.id}`}>
                <h3 className='fonty' > Id#: {game.id}</h3><br/>
                <p className='fonty'> Game Name: {game.name}</p><br />
                <p className='fonty'> Max players: {game.maxplayers}</p><br />
                <p className='fonty'> Location: {game.location}</p><br />
                <p className='fonty'> {moment(`${game.date}`).format('lll')}</p>
                <button className='delBut' onClick={e => deleteGames(game.id)}> Delete </button>

            </div >))}





        </div>
    )
}

export default Games;