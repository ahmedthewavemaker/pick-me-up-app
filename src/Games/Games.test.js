import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Games from './Games';
import { MemoryRouter } from 'react-router-dom';

describe('Login component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter> <Games />
      </MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter> <Games  />
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})