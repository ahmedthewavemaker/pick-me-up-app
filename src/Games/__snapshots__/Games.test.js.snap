// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders the UI as expected 1`] = `
Array [
  " ",
  <div
    className="gameview"
  />,
]
`;
